platform/engineering/design-docs/2021-01-13-web-component-library.md

```diff
@@ -0,0 +1,185 @@
+# Web Component library
+
+**Author(s):** Brooks Johnson  
+**Last Updated:** January 13, 2021  
+**Status:** **Draft** | In Review | Approved  
+**Approvers:** _Chris Valarida_ \[ \], _Michael Fleet_ \[ \]
+
+
+## Overview
+
+### Objective
+
+The goal is to establish a component library based on [Web Components](https://developer.mozilla.org/en-US/docs/Web/Web_Components), which are a browser standard. This will eventually supercede the current `component-library` code and some of the liquid HTML templates currently in `vets-website`. A non-goal is creating an entirely new visual experience as part of the component library.
+
+The intended audience for this document is frontend engineers.
+
+### Background
+
+Currently, the design system has various implementations of "components" scattered in various places to serve various tools/libraries:
+
+- The `formation` npm package
+    - This contains static HTML/CSS (and sometimes JS), and is used by copy-pasting markup onto static pages
+-  The `component-library` package
+    - This is the rebranded `formation-react`, containing React implementations of some patterns defined in `formation`. Can only be used inside React apps
+- [Liquid templates in `vets-website`](https://github.com/department-of-veterans-affairs/vets-website/tree/958d89f18db29a1f3e1f1cbe4e304299eb387de7/src/site)
+    - This is similar to the static markup found in `formation`. These templates of components are only used in the content build for va.gov
+
+Maintaining multiple sources of truth is a burden for all teams involved and leads to wasted time.
```

Comment:
Tell me how using Web Components is going to solve this problem.

Reply:
From the design doc template:

> The background section should contain information the reader needs to know to understand the problem being solved. This can be a combination of text and links to other documents.
>
> Do NOT describe the solution here. That goes in High Level Design.

Web Components are part of the solution, so should I really mention that here?

Reply:
No, you're right. I was looking for the first sentence of the `High Level Design` you've already got. (Literally the sentence after this one. :facepalm:) I completely skipped over it.

>Maintaining multiple sources of truth is a burden...

>The goal is to establish a single codebase and documentation for the implementation of design system components.

You've got it covered. :+1: 

Bonus points for using the same terminology, though. Maybe something like

> Maintaining multiple codebases and sources of documentation for effectively the same components is a burden...